src/supervisor/watchers/handlers/argo-rollout.ts

```diff
@@ -0,0 +1,215 @@
+import { IncomingMessage } from 'http';
+import { deleteWorkload } from './workload';
+import { WorkloadKind } from '../../types';
+import {
+  FALSY_WORKLOAD_NAME_MARKER,
+  V1alpha1Rollout,
+  V1alpha1RolloutList,
+} from './types';
+import { paginatedClusterList, paginatedNamespacedList } from './pagination';
+import { k8sApi } from '../../cluster';
+import {
+  deleteWorkloadAlreadyScanned,
+  deleteWorkloadImagesAlreadyScanned,
+  kubernetesObjectToWorkloadAlreadyScanned,
+} from '../../../state';
+import { retryKubernetesApiRequest } from '../../kuberenetes-api-wrappers';
+import { logger } from '../../../common/logger';
+import { deleteWorkloadFromScanQueue } from './queue';
+import { trimWorkload } from '../../workload-sanitization';
+
+export async function paginatedNamespacedArgoRolloutList(
+  namespace: string,
+): Promise<{
+  response: IncomingMessage;
+  body: V1alpha1RolloutList;
+}> {
+  const rolloutList = new V1alpha1RolloutList();
+  rolloutList.apiVersion = 'argoproj.io/v1alpha1';
+  rolloutList.kind = 'RolloutList';
+  rolloutList.items = new Array<V1alpha1Rollout>();
+
+  return await paginatedNamespacedList(
+    namespace,
+    rolloutList,
+    async (
+      namespace: string,
+      pretty?: string,
+      _allowWatchBookmarks?: boolean,
+      _continue?: string,
+      fieldSelector?: string,
+      labelSelector?: string,
+      limit?: number,
+    ) =>
+      k8sApi.customObjectsClient.listNamespacedCustomObject(
+        'argoproj.io',
+        'v1alpha1',
+        namespace,
+        'rollouts',
+        pretty,
+        _continue,
+        fieldSelector,
+        labelSelector,
+        limit,
+        // TODO: Why any?
```

Comment:
wot dis

Reply:
It's because:

> Type 'Promise<{ response: IncomingMessage; **body: object**; }>' is not assignable to type 'Promise<{ response: IncomingMessage; **body: KubernetesListObject<KubernetesObject & Partial<{ status: unknown; spec: unknown; }>>**; }>'

The K8s client's `listNamespacedCustomObject()` doesn't allow to specify the type of the response body and returns the generic `object`, but with how we declared our types we expect it to return a `KubernetesListObject`